crates/pallets/starknet/src/lib.rs

```diff
@@ -378,10 +378,21 @@ pub mod pallet {
             }
 
             for (sierra_class_hash, casm_class_hash) in self.sierra_to_casm_class_hash.iter() {
+                assert!(
+                    ContractClasses::<T>::contains_key(casm_class_hash),
```

Comment:
isn't this suppoed to be Sierra class hash?

Reply:
You're right, 'class hashes' typically refer to Sierra hashes as outlined [here](https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/class-hash/). However, in the genesis.json file within the test folder, the only contract class hash listed under `sierra_class_hash_to_casm_class_hash` (0x02f99bf9799ada84cd5ac0d0fe36b9d8f65efcb377cd2e8cf8309ad2daf15e4b) is actually a CASM hash, according to that mapping. This led to my confusion.
Is it a mistake so I can change that file or does it have some special meaning there?

Reply:
changed to sierra hash

Reply:
They are now, I am not sure if it's correct though. I can change the behavior of this check to succeed either the sierra or casm hash from the mapping is in the `ContractClasses`, but not sure if it should be like this

Reply:
> the only contract class hash listed under sierra_class_hash_to_casm_class_hash

I didn't get this @kfastov, based on the naming it appears arr[0] is sierra class hash and arr[1] is casm class hash, are you implying that's not the case? If yes, do you know what those hashes are?

Reply:
> So the keys in ContractClasses can be Casm or Sierra ContractClassHash?

No @tdelabro, they should be sierra class hashes only. The only place we use Casm class hash should be `CompiledClassHashes` afaik,

Reply:
@apoorvsadana this won't work for cairo0 contracts. They don't have sierra, they can't have a sierra class hash

Reply:
Oh ok, ya in that context it would casm, understood.

Reply:
The design decision we are facing now is:

1)
ContractClasses: k:CasmContractClassHash v: CasmContractClass
To execute cairo0, directly index ContractClasses
To execute cairo-1 index  `CompiledClassHashes` to get the casm hash then index the `ContractClasses` with it

2)
ContractClasses: k: CasmContractClassHash |  SierraContractClassHash v: CasmContractClass
To execute bot cairo0 and cairo1, directly index `ContractClasses`

Reply:
1 is cleaner but 2 is probably more performant coz it spares one read.
But it's kinda negligible imo

Reply:
@apoorvsadana @antiyro @EvolveArt wdyt?

Reply:
2 looks way better to me. Why negligible we have a linear improvement no?

Reply:
Works for me.

@kfastov can you go solution 2?
Maybe with some type aliasing so the name of the types used as key is more explicit. `SierraOrCasmClassHash`

Reply:
Do I have this right:

1) The keys in `ContractClasses` include both Sierra and Casm hashes.
2) This is also the case for `contract_classes` in the genesis.json file.
3) Therefore, when validating the Sierra to Casm mapping in `genesis.json`, we should ensure that only one of the hashes (either Sierra or Casm) is present in `ContractClasses`.
4) Once this mapping is validated, the corresponding hash (whichever is not already present) should also be added. This allows for indexing the same class using both types of hashes.

Reply:
@apoorvsadana Can I mark it as resolved?